pcsx2/gui/AppUserMode.cpp

```diff
@@ -27,284 +29,344 @@
+		try
+		{
+			std::ifstream in = getFileAsInputStream(fileName);
+			stream = YAML::Load(in);
+			in.close();
+			std::ostringstream os;
+			os << stream;
+			data = os.str();
+			return true;
+		}
+		catch (const std::exception& e)
+		{
+			DevCon.Warning("ERROR: ", e.what());
+			data = "";
+			return false;
+		}
+	}
+	else
+	{
+		return false;
+	}
+}
 
-		FirstTimeWizard wiz( NULL );
-		if( wiz.RunWizard( wiz.GetFirstPage() ) ) break;
-		if (wiz.GetReturnCode() != pxID_RestartWizard)
-			throw Exception::StartupAborted( L"User canceled FirstTime Wizard." );
+void Pcsx2App::Save(fs::path fileName)
+{
+	if (!Path::DoesExist(fileName.parent_path().make_preferred()))
+	{
+		Path::CreateFolder(fileName.parent_path().make_preferred());
+	} 
 
-		Console.WriteLn( Color_StrongBlack, "Restarting First Time Wizard!" );
+	if (stream)
+	{
+		try
+		{
+			std::ofstream out = getFileAsOutputStream(fileName);
+			out << stream;
+			out.close();
+		}
+		catch (const std::exception& e)
+		{
+			DevCon.Warning("ERROR: ", e.what());
+		}
 	}
 }
 
-wxConfigBase* Pcsx2App::OpenInstallSettingsFile()
+bool Pcsx2App::OpenInstallSettingsFile()
 {
 	// Implementation Notes:
 	//
-	// As of 0.9.8 and beyond, PCSX2's versioning should be strong enough to base ini and
+	// As of 0.9.8 and beyond, PCSX2's versioning should be strong enough to base yaml and
 	// plugin compatibility on version information alone.  This in turn allows us to ditch
-	// the old system (CWD-based ini file mess) in favor of a system that simply stores
+	// the old system (CWD-based yaml file mess) in favor of a system that simply stores
 	// most core application-level settings in the registry.
 
-	std::unique_ptr<wxConfigBase> conf_install;
+	InstallationMode = InstallationModeType::InstallMode_Registered;
+	fs::path usrlocaldir = PathDefs::GetUserLocalDataDir();
 
-#ifdef __WXMSW__
-	conf_install = std::unique_ptr<wxConfigBase>(new wxRegConfig());
-#else
-	// FIXME!!  Linux / Mac
-	// Where the heck should this information be stored?
+	InstallFolder = (wxFileName(wxStandardPaths::Get().GetExecutablePath())).GetPath().ToStdString();
+
+	std::string usermodeFile = (GetAppName().ToStdString() + "-reg.yaml");
+	usermodePath = Path::Combine(usrlocaldir, usermodeFile); 
 
-	wxDirName usrlocaldir( PathDefs::GetUserLocalDataDir() );
-	//wxDirName usrlocaldir( wxStandardPaths::Get().GetDataDir() );
-	if( !usrlocaldir.Exists() )
+	if (!Path::DoesExist(usermodePath))
 	{
-		Console.WriteLn( L"Creating UserLocalData folder: " + usrlocaldir.ToString() );
-		usrlocaldir.Mkdir();
+		CustomDocumentsFolder = PathDefs::AppRoot();
+
+		// Install Mode, genereate a new stream and manually write data to new file if it doesn't.
+		stream["DocumentsFolderMode"] = (int)DocsFolderMode;
+		stream["CustomDocumentsFolder"] = CustomDocumentsFolder.string();
+		stream["UseDefaultSettingsFolder"] = UseDefaultSettingsFolder;
+		stream["SettingsFolder"] = SettingsFolder.string();
```

Comment:
Marking as needing to verify - is this a lossy conversion -- saving/restoring as a std::string in windows?

Reply:
Not a lossy conversion, tested both ways.  Storing a path with unicode into the YAML file, and grabbing the path out of the file into the `fs::path` variable.  Looks good to me.